src/middleware/devtools.ts

```diff
@@ -173,22 +202,54 @@ const devtoolsImpl: DevtoolsImpl =
       return fn(set, get, api)
     }
 
-    const extension = extensionConnector.connect(options)
+    const { connection, ...connectionInformation } = (() => {
+      // For backwards-compatibility's sake, we don't track
+      // a store unless it's given us a `store` string
+      if (store === undefined) {
+        return {
+          type: 'untracked' as const,
+          connection: extensionConnector.connect(options),
+        }
+      }
+      const existingConnection = trackedConnections.get(options.name)
+      if (existingConnection) {
+        return { type: 'tracked' as const, store, ...existingConnection }
+      }
+      const newConnection: ConnectionInformation = {
+        connection: extensionConnector.connect(options),
+        stores: {},
+      }
+      trackedConnections.set(options.name, newConnection)
+      return { type: 'tracked' as const, store, ...newConnection }
+    })()
```

Comment:
Are you sure we need IIFE here?

Reply:
Extracted this into a helper function in another PR.